framework/src/controller/bus.ts

```diff
@@ -392,31 +408,50 @@ export class Bus {
 	private async _setupIPCServer(): Promise<void> {
 		await this._ipcServer.start();
 
-		this._ipcServer.rpcServer.expose(
-			'registerChannel',
-			(moduleAlias, events, actions, options, cb: NodeCallback) => {
-				this.registerChannel(moduleAlias, events, actions, options)
-					.then(() => cb(null))
-					.catch(error => cb(error));
-			},
-		);
-
-		this._ipcServer.rpcServer.expose(
-			'invoke',
-			(message: string | JSONRPC.RequestObject, cb: NodeCallback) => {
-				this.invoke(message)
-					.then(data => {
-						cb(null, data as JSONRPC.ResponseObjectWithResult);
+		const listenToEvents = async () => {
+			for await (const [_event, eventData] of this._ipcServer.subSocket) {
+				this.publish(eventData.toString());
+			}
+		};
+
+		listenToEvents()
+			.catch(err => {
+				this.logger.debug(err, 'Error occured while listening to events on subscriber.');
+			});
+
+		const listenToRPC = async () => {
+			for await (const [sender, request, params] of this._ipcServer.rpcServer) {
+				if (request.toString() === IPC_REGISTER_CHANNEL_EVENT) {
+					const { moduleAlias, eventsList, actionsInfo, options } = JSON.parse(params.toString()) as RegisterToBusRequestObject;
+					this.registerChannel(moduleAlias, eventsList, actionsInfo, options);
+					continue;
+				}
+				if (request.toString() === IPC_RPC_EVENT) {
+					const requestData = JSON.parse(params.toString()) as JSONRPC.RequestObject;
+					this.invoke(requestData)
+						.then(result => {
+							// Send back result RPC request for a given requestId
+							this._ipcServer.rpcServer.send([
```

Comment:
Does not this be handled in the `this.invoke`?

Reply:
we will have to also pass `sender` detail to `invoke` so trying to avoid that. Also, this listener is called `listenToRPC` so maybe its better to also handler it here